sdk/python/lib/pulumi/asset.py

```diff
@@ -36,10 +37,11 @@ class FileAsset(Asset):
     A FileAsset is a kind of asset produced from a given path to a file on
     the local filesysetm.
     """
-    def __init__(self, path: str) -> None:
-        if not isinstance(path, str):
-            raise TypeError("FileAsset path must be a string")
-        self.path = path
+
+    def __init__(self, path: Union[str, PathLike]) -> None:
+        if not isinstance(path, (str, PathLike)):
+            raise TypeError("FileAsset path must be a string or os.PathLike")
+        self.path = str(path)
```

Comment:
Minor - but I think technically it’s jot a guarantee that PathLike coerces correctly under `str()`. I suspect this should be `os.fspath()`?

Reply:
Oh thanks, I didn't even think about that since `pathlib.Path` they're the same...

I'll update this.

Reply:
Fixed.